viewing_party/party.py

```diff
@@ -1,23 +1,201 @@
+        if movie["title"] not in friend_titles_watched:
+            unique_watched.append(movie)
+
+    return unique_watched
+
+def get_friends_unique_watched(user_data):
+    friends_watched = []
+    friends_unique_watched = []
+    user_titles = set()
+
+    # iterate through user list
+       # append title to user_watched
+    for movie in user_data["watched"]:
+        user_titles.add(movie["title"])
+
+    # iterate through user_data of friends
+       # if movie["title"] is not in user_titles
+           # append movie to friends_unique_watched
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["title"] not in user_titles:
+                friends_watched.append(movie)
+
+    for i in range(len(friends_watched)):
+        if friends_watched[i] not in friends_watched[i+1:]:
+            friends_unique_watched.append(friends_watched[i])
+
+    return friends_unique_watched
+
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    recommended_movies = []
+    friends_watched = get_friends_unique_watched(user_data)
+
+    # loop through friends unique watched
+    for movie in friends_watched:
+        # if the movie is in subscriptions list
+        if movie["host"] in user_data["subscriptions"]:
+            # add it to the output list
+            recommended_movies.append(movie)
+
+    return recommended_movies
+
+
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+'''
+There are four tests about a get_new_rec_by_genre function
+Create a function named get_new_rec_by_genre
+
+takes one parameter: user_data
+Consider the user's most frequently watched genre. Then, determine a list of recommended movies.
+A movie should be added to this list if and only if:
+The user has not watched it
+At least one of the user's friends has watched
+The "genre" of the movie is the same as the user's most frequent genre
+Return the list of recommended movies
+
+'''
+
+'''
+There are also two tests about a get_rec_from_favorites function
+Create a function named get_rec_from_favorites
+
+takes one parameter: user_data
+user_data will have a field "favorites". The value of "favorites" is a list of movie dictionaries
+This represents the user's favorite movies
+Then, determine a list of recommended movies. A movie should be added to this list if and only if:
+The movie is in the user's "favorites"
+None of the user's friends have watched it
+Return the list of recommended movies
+'''
+
+
+def get_new_rec_by_genre(user_data):
+    recommended_movies = []
+    friends_watched = get_friends_unique_watched(user_data)
+    most_watched_genre = get_most_watched_genre(user_data)
+
+    for movie in friends_watched:
+        if movie["genre"] == most_watched_genre:

```

Comment:
I see your tests passed for all the waves :)

I did want to call out that a movie should be added to recommended_movies if the **user has not watched it**, at least 1 friend has watched it, and the genre is the user's most frequent genre. 

You could check if the movie has not been watched by the user by adding a check on line 184, maybe something like:

`if movie["genre"] == most_watched_genre and movie not in user_data["watched"]:`